cmd/src/debug_comp.go

```diff
@@ -0,0 +1,228 @@
+	}
+
+	debugCommands = append(debugCommands, &command{
+		flagSet: flagSet,
+		handler: handler,
+		usageFunc: func() {
+			fmt.Println(usage)
+		},
+	})
+}
+
+// writes archive of common docker cli commands
+func archiveCompose(ctx context.Context, zw *zip.Writer, verbose, noConfigs bool, baseDir string) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	containers, err := getContainers(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to get docker containers")
+	}
+
+	if verbose {
+		log.Printf("getting docker data for %d containers...\n", len(containers))
+	}
+
+	// setup channel for slice of archive function outputs
+	ch := make(chan *archiveFile)
+	g, ctx := errgroup.WithContext(ctx)
+	semaphore := semaphore.NewWeighted(8)
+
+	run := func(f func() *archiveFile) {
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+
+			if file := f(); file != nil {
+				ch <- file
+			}
+
+			return nil
+		})
+	}
+
+	// start goroutine to run docker ps -o wide
+	run(func() *archiveFile { return getPs(ctx, baseDir) })
+
+	// start goroutine to run docker container stats --no-stream
+	run(func() *archiveFile { return getStats(ctx, baseDir) })
+
+	// start goroutine to run docker container logs <container>
+	for _, container := range containers {
+		container := container
+		run(func() *archiveFile { return getContainerLog(ctx, container, baseDir) })
+	}
+
+	// start goroutine to run docker container inspect <container>
+	for _, container := range containers {
+		container := container
+		run(func() *archiveFile { return getInspect(ctx, container, baseDir) })
+	}
+
+	// start goroutine to get configs
+	if !noConfigs {
+		run(func() *archiveFile { return getExternalServicesConfig(ctx, baseDir) })
+
+		run(func() *archiveFile { return getSiteConfig(ctx, baseDir) })
+	}
+
+	// close channel when wait group goroutines have completed
+	go func() {
+		if err := g.Wait(); err != nil {
+			fmt.Printf("archiveCompose failed to open wait group: %s\n", err)
+			os.Exit(1)
+		}
+		close(ch)
+	}()
+
+	// Read binaries from channel and write to archive on host machine
+	if err := writeChannelContentsToZip(zw, ch, verbose); err != nil {
+		return errors.Wrap(err, "failed to write archives from channel")
+	}
+
+	return nil
+}
+
+// Returns list of containers that are members of the docker-compose_sourcegraph
+func getContainers(ctx context.Context) ([]string, error) {
+	c, err := exec.CommandContext(ctx, "docker", "container", "ls", "--format", "{{.Names}} {{.Networks}}").Output()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get container names with error")
+	}
+	s := string(c)
+	preprocessed := strings.Split(strings.TrimSpace(s), "\n")
+	containers := make([]string, 0, len(preprocessed))
+	for _, container := range preprocessed {
+		tmpStr := strings.Split(container, " ")
+		if len(tmpStr) >= 2 && tmpStr[1] == "docker-compose_sourcegraph" {
```

Comment:
Does `docker-compose_sourcegraph` need to be a flag the user can customise? How much variety do we see in Docker Compose environment names?

Reply:
@DaedalusG Could we create a follow-up issue for this one?

Reply:
I don't think we've ever seen a customer customize the compose network naming conventions, but it could be out there in the wild. Definitely worth adding a flag to allow users to supply their own value.